src/main/java/uk/gov/justice/services/test/domain/EventsChecker.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.justice.services.test.domain;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.equalToIgnoringCase;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static uk.gov.justice.services.test.DomainTest.eventNameFrom;
+import static uk.gov.justice.services.test.DomainTest.generatedEventAsJsonNode;
+import static uk.gov.justice.services.test.DomainTest.jsonNodeWithoutMetadataFrom;
+
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.lang3.StringUtils;
+
+public class EventsChecker {
+
+    public static void compareEvents(List<JsonNode> expectedEvents, List<Object> actualEvents) {
+
+        assertThat(createMessage(expectedEvents, actualEvents, null), expectedEvents.size(), is(actualEvents.size()));
+
+        int index = 0;
+        for (JsonNode jsonNode : expectedEvents) {
+
+            final Object generatedEvent = actualEvents.get(index);
+            assertThat(createMessage(expectedEvents, actualEvents, eventNameFrom(generatedEvent)), eventNameFrom(generatedEvent), equalToIgnoringCase(eventNameFrom(jsonNode)));
+            assertThat(createMessage(expectedEvents, actualEvents, null), generatedEventAsJsonNode(generatedEvent), equalTo(jsonNodeWithoutMetadataFrom(jsonNode)));
+            index++;
+        }
+    }
+
+    private static String createMessage(List<JsonNode> expectedEvents, List<Object> generatedEvents, String missingEvent) {
+        final StringBuilder buffer = new StringBuilder();
+        if (StringUtils.isNotEmpty(missingEvent)) {
+            buffer.append("Event not found ").append(missingEvent).append(System.lineSeparator());
```

Comment:
Part of the purpose of this change is to make the error message as useful as possible when the events do not match. I don't think that "Event not found" is very helpful, especially since I think you're printing the name of the actual event, not the expected event.

As a developer using this tool, I would like to see a clear output that highlights what exactly was wrong with the actual events - eg, I'm not interested in events in the list that matched, only in the ones that didn't. And I'd like to be able to see clearly if it was the wrong event entirely (eg, the name was different) or whether the payload didn't match.